contributed_definitions/nyaml/NXcoordinate_system_set.yaml

```diff
@@ -88,138 +88,259 @@ doc: |
-#          :ref:`NXcoordinate_system_set` instance using instances of NXcoordinate_system.
-#          
-#          How many individual instances of NXcoordinate_system to allow within one
-#          instance of :ref:`NXcoordinate_system_set`?
-#          
-#          * 0; This case should be avoided for the sake of clarity but this case could
-#            mean the authors of the definition meant that McStas is used. We conclude,
-#            McStas is used in this case.
-#          * 1; Like above-mentioned this case has the advantage that it is explicit
-#            and faces no ambiguities. However, in reality typically multiple
-#            coordinate systems have to be mastered especially for complex
-#            multi-signal modality experiments.
-#          * 2 or more; If this case is realized, the best practice is that in every
-#            case where a coordinate system should be referred to the respective class
-#            has a depends_on field which resolves the possible ambiguities which specific
-#            coordinate systems is referred to. The benefit of this explicit and clear
-#            specifying of the coordinate system used in every case is that especially
-#            in combination with having coordinate systems inside deeper branches
-#            makes up for a very versatile, backwards compatible, but powerful system
-#            to express different types of coordinate systems using NeXus. In the case
-#            of two or more instances of NXcoordinate_system in one :ref:`NXcoordinate_system_set`,
-#            it is also advised to specify the relationship between the two coordinate systems by
-#            using the (NXtransformations) group within NXcoordinate_system.
-#          
-#          In effect, 1 should be the preferred choice. However, if more than one coordinate
-#          system is defined for practical purposes, explicit depends_on fields should
-#          always guide the user for each group and field which of the coordinate system
-#          one refers to.
-#     </doc>
-#     <group type="NXcoordinate_system"/>
-#     <!--How to interpret "depends_on" calls for NXtransformations instances which live
-# within an instance of NXcoordinate_system(_set)?
-# Depends on cardinality of NXcoordinate_system_set, i.e. how many NXcoordinate_system are defined?
-# - 0, the root of an NXtransformation chain i.e. "." means McStas
-# - 1, the root of an NXtransformation chain i.e. "." means "that specific CS" i.e. the one defined by the single instance of NXcoordinate_system in the set
-# - > 1, "." becomes ambiguous. My suggestion, spell out the name of the specific NXcoordinate_system instance to be used, using "."
-# despite all these, a warning should be raised, like a logic warning (and McStas kick in) or an error (i.e. be forbidden) be raised-->
-# </definition>
+  # frequently used coordinate systems with frequently relevant specializations
```

Comment:
I am not a big fan of having these explicitly named coordinate systems here? What do we gain from that? They should be defined on the appdef level.

Reply:
To be able to used them just as they are defined (see my previous point) avoid creating conceptual inconsistencies during appdef writing when docstrings are not fully copied over etc. etc, like building blocks and the more I think about it again I feel the idea of introducing some bridge between NeXus = set of super generic base classes + { {super specific base classes + specializations + constraints }} i.e. appdef would be useful like sub-parts of concept graphs that a community can reuse which are stronger specialized than base classes but can be reutilized for composing them conveniently into appdefs. Unfortunately in NeXus this does not exist, in OWL this could be possible but current NFDI activities go the opposite way they just create data schemas using OWL + the associated tools from the semantic web technology stack without ontology matching as the key priority and optimization aim

Reply:
Although my answer is tangible and to the point of NeXus' key weakness, a practical solution to go forward is indeed to move these specific coordinate system concepts (you on point here these are specific instances of concepts) but I do not want to repeat their writing in now NXapm, NXem, NXapm_PARAPROBE_TOOLS, NXapm_compositionspace_TOOLS, NXmicrostructure_CLASSES, mainly to avoid creating inconsistencies and having to copy over things which should be exactly the same, single source principles except them embedded in a different parent graphs' namespace.

Reply:
Ok, now I think I got your main point, namely that you want to have a place for conventions that are used across many (but of course not all) different domains, but you don't want to have them in the appdefs because you would need to repeat them a lot of times. Got it.

My suggestion for compromise would be the following: keep the specific NXcoordinate_systems here, but remove all of the unneeded repititions from the base class `NXcoordinate_system` (like handedness, and so on), and then remove anything that is strictly speaking domain-specific (like mentioning of a prefered direction, an electron beam, and so on). Then we can see if it is worth it to keeps these explicitly named coordinate systems (so they can be reused in em, apm, ...), without polluting the base class too much.

Reply:
Exactly this was my point. Indeed, speculatively I followed your suggestion and pushed it as cc18a27b